pkg/api/resource/quantity.go

```diff
@@ -190,7 +190,9 @@ func ParseQuantity(str string) (*Quantity, error) {
 	// of an amount.
 	// Arguably, this should be inf.RoundHalfUp (normal rounding), but
 	// that would have the side effect of rounding values < .5m to zero.
-	amount.Round(amount, 3, inf.RoundUp)
+	if v, ok := amount.Unscaled(); v != int64(0) && ok {
```

Comment:
Thinking about this more-- if you look down to line 266 ```exponent := int(-q.Amount.Scale())``` seems to be the culprit, if the value is 0 then the scale should be ignored.

I think this change here is good, but you should still be able to trigger the invalid behavior by constructing a value, so maybe you can fix the output, too?

(e.g., construct a value of "1000m", then use inf package to subtract 1000 from it and construct a new Quantity with that; this should trigger the incorrect output behavior.)